causync.py

```diff
@@ -270,11 +309,10 @@ def rmtree(self, dirnames):
             except FileNotFoundError:
                 pass
 
+    def create_lockfile(self, path):
+        """ Creates a lockfile at the specified path. """
```

Comment:
Please make this safer. The lock file should not exist, so throw an exception if it does. Then open the file for writing with using mode 'w' instead of 'a'. And please write the PID of the current process into this file, so later one can check whether this lock is still active or dead.